### PR TITLE
[fix] qr code scanner bug fixes

### PR DESCRIPTION
**Pull request by stephaniewong2**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

QR Code:
- Created a separate bottom tab option for the qr code scanner.
- Migrated to the expo camera library.
- Added green borders and toast message on valid scans.
- Added red borders on invalid scans.
- Moved toasts to top of the screen and improved misc usability.
- Rewrote queries to fetch case data more efficiently.

Case Screen:
- Removed 3 dots from `CaseCard`.
- Fixed `AllCases` to span entire screen.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

  

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

- Check that `AllCases` screen looks proper with a varying number of cases. Might need to account for bottom tab in setting height.
- Use the qr code scanner on a variety of codes and verify that the correct output occurs.

### Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

[Expo Camera Documentation](https://docs.expo.dev/versions/latest/sdk/camera/#ratio)
[Debouncing Functions](https://medium.com/@rameshsainom/use-of-throttling-and-debouncing-in-react-native-3bdac01a80c3)
[Lodash Library for Debouncing](https://www.npmjs.com/package/lodash)

CC: @ronniebeggs

### CODE REVIEW

File: src/app/(BottomTabNavigation)/AllCases/QRCodeScanner/styles.ts [hunk 1]
```diff
@@ -21,10 +21,16 @@ export default StyleSheet.create({
     height: 300,
     width: 300,
     border: 'solid',
-    borderWidth: 2,
+    borderWidth: 3,
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/QRCodeScanner/styles.ts]: Is there any way we can have more round edges for the little camera window in the barcode scanner?

File: src/app/(BottomTabNavigation)/AllCases/QRCodeScanner/index.tsx [hunk 2]
```diff
@@ -121,7 +121,11 @@ function QRCodeScannerScreen() {
       <Text style={styles.topText}>Point your Camera at the QR code.</Text>
       <BarCodeScanner
         onBarCodeScanned={handleBarCodeScanned}
-        style={[styles.scanner]}
+        style={
+          scanned
+            ? [styles.scanner, styles.scanned]
+            : [styles.scanner, styles.notScanned]
+        }
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/QRCodeScanner/index.tsx]: In the same way that the border turns screen when it's valid, would it be possible to turn red when invalid?

**Review by ronniebeggs — CHANGES_REQUESTED** [review]
I like the changes so far. I'm considering a few edge cases that we haven't decided how to handle yet.

When a user scans a valid barcode, the view case option will pop up and the border will turn green. However, if a different barcode is scanned right after, the screen should update:
- If the other barcode is valid, it'd be nice to somehow indicate that a different case was now scanned.
- If the other barcode is invalid, the view case option should disappear while the error toast message pops up. Maybe the border should turn red as well.

It also might be nice to move toast messages to the top of the screen so they don't overlap with the view case button.

**Comment by ronniebeggs** [comment]
Also..... BarCodeScanner is lowkey being deprecated: https://docs.expo.dev/versions/latest/sdk/bar-code-scanner/#component

**Comment by ronniebeggs** [comment]
Josh Notes:
- new qr codes should be processed when looked at
- red border for invalid codes
- toast message for valid codes
- move all toast messages (valid and invalid) to the top of the screen

**Review by ronniebeggs — CHANGES_REQUESTED** [review]
I basically rewrote the qr code query logic so it might look like a lot. Can you double check that the qr code works for different scanning scenarios?
